### PR TITLE
feature: Saving current scatterplot to the URL

### LINKED ISSUE #182: Save Scatter Plot settings to URL (+current open tab)
## Use Case
- Save the x and y features and the current range type of the scatter plot to the URL so that it can be shared.
- Also save the currently open tab, for scientists sharing scatter plot views.

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #182! Saves the currently open tab and the scatterplot configuration to the URL.

*Expected review size: medium, ~20-30 minutes*

Solution
========
- Adds serializing and deserializing for the `ScatterPlotConfig`, allowing it to be saved and retrieved from the URL.
- Adds the currently open tab to the `ViewerConfig`, and saves it to and from the URL.

## Type of change

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-233/?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Fcollection.json&dataset=Mama%20Bear&feature=Volume&track=602&t=123&color=matplotlib-cool&palette-key=adobe&bg-sat=100&bg-brightness=100&fg-alpha=100&outlier-color=c0c0c0&outlier-mode=1&filter-color=dddddd&filter-mode=1&tab=scatter_plot&scalebar=1&timestamp=1&path=1&keep-range=0&scatter-range=frame&scatter-x=Height&scatter-y=Volume
2. The scatterplot tab should already be open, with "Height" and "Volume" as the two axes. The range should be set to "Current frame."

image

Thanks for contributing!

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -119,7 +120,6 @@ function App(): ReactElement {
   }, [dataset, featureThresholds]);
 
   const [playbackFps, setPlaybackFps] = useState(DEFAULT_PLAYBACK_FPS);
-  const [openTab, setOpenTab] = useState(TabType.TRACK_PLOT);
```
Comment (ShrimpCryptid) [inline src/App.tsx:122]: `openTab` is now part of `ViewerConfig` (`config`)

File: src/colorizer/types.ts [hunk 2]
```diff
@@ -134,6 +134,18 @@ export type DrawSettings = {
   color: Color;
 };
 
+// CHANGING THESE VALUES CAN POTENTIALLY BREAK URLs. See `url_utils.parseDrawSettings` for parsing logic.
+export enum TabType {
+  FILTERS = "filters",
+  TRACK_PLOT = "track_plot",
+  SCATTER_PLOT = "scatter_plot",
+  SETTINGS = "settings",
+}
```
Comment (ShrimpCryptid) [inline src/colorizer/types.ts:143]: There is a matching unit test that checks for these explicit string values, to make sure they don't accidentally get changed in the future.

**Review by kmitcham — APPROVED** [review]
Looks good to me.  What is the correct/expected behavior if I go to a plot based on a URL, and update the plot values on the page? Should the URL update, or just not be in sync with the display until serialize is called?

**Comment by ShrimpCryptid** [comment]
> Looks good to me. What is the correct/expected behavior if I go to a plot based on a URL, and update the plot values on the page? Should the URL update, or just not be in sync with the display until serialize is called?

The URL should always update in sync with changes on the page. The serialization of the current config is triggered [here](https://github.com/allen-cell-animated/nucmorph-colorizer/pull/233/files#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R224-R225), where any changes to `scatterPlotConfig` causes the URL updating code to fire.

**Review by ascibisz — APPROVED** [review]
Code looks good to me! I'm glad you wrote tests, I also tested manually and everything worked as expected for me

**Pull request merged**